### PR TITLE
Add the Meld prototype framework scheduler

### PR DESCRIPTION
**Pull request by knoepfel**
This PR supports Meld (https://github.com/knoepfel/meld)

### CODE REVIEW

File: packages/meld/package.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#
+#     spack install meld
+#
+# You can edit this file again by typing:
+#
+#     spack edit meld
+#
+# See the Spack documentation for more information on packaging.
+# ----------------------------------------------------------------------------
+
```
Comment (greenc-FNAL) [inline packages/meld/package.py]: Remove boilerplate.

File: packages/meld/package.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    version("develop", branch="main", get_full_repo=True)
+
+    cxxstd_variant("20", "23", default="20", sticky=True)
+
+    depends_on("boost@1.75.0: +json+program_options+stacktrace")
+    depends_on("fmt@:9")
+    depends_on("jsonnet")
+    depends_on("spdlog")
+    depends_on("intel-tbb")
```
Comment (greenc-FNAL) [inline packages/meld/package.py]: Please refer to the virtual package (`tbb`) instead of the concrete: we usually use `intel-oneapi-tbb`.

**Review by greenc-FNAL — CHANGES_REQUESTED** [review]

File: packages/meld/package.py (hunk 1 above)
Comment (knoepfel) [reply packages/meld/package.py]: Done.

File: packages/meld/package.py (hunk 2 above)
Comment (knoepfel) [reply packages/meld/package.py]: Unfortunately, I can't use `intel-oneapi-tbb`.  Both `intel-tbb` and `intel-oneapi-tbb` are virtual packages that provide `tbb`.  The difference is that `intel-oneapi-tbb` runs an installation script that cannot be used on macOS Arm systems, whereas `intel-tbb` does support building from source code.

Comment (knoepfel) [reply packages/meld/package.py]: @greenc-FNAL, I think I got the sign wrong: `tbb` is the virtual package, and both `intel-oneapi-tbb` and `intel-tbb` are the concrete ones.  So maybe the recommendation is to switch to just `tbb`?

Comment (greenc-FNAL) [reply packages/meld/package.py]: I would claim that the inability of `intel-oneapi-tbb` to satisfy the requirement on macOS/arm64 is a packaging/platform issue, and should be managed by an entry in the appropriate `packages.yaml` overriding the virtual package priorities for that platform. The maximum flexibility should be retained in the recipe itself.

Comment (knoepfel) [reply packages/meld/package.py]: Thanks for the explanation, Chris.  Recipe adjusted accordingly.

**Review by greenc-FNAL — APPROVED** [review]

**Pull request merged**